src/hooks/useOnClickOutside.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react';
+
+export const useOnClickOutside = (
+  ref: React.RefObject<HTMLElement>,
+  handler: (event: MouseEvent | TouchEvent) => void,
+) => {
+  useEffect(() => {
+    const eventListener = (event: MouseEvent | TouchEvent) => {
+      if (ref.current && !ref.current.contains(event.target as Node)) {
+        handler(event);
+      }
+    };
+
+    document.addEventListener('mousedown', eventListener);
+    document.addEventListener('touchstart', eventListener);
```

Comment:
Events stay registered the whole time component is alive thus event listeners will be called every time user clicks somewhere.